### PR TITLE
Replace all occurrences of `size_t` with `std::size_t`

### PR DESCRIPTION
**Pull request by aalkin**

### CODE REVIEW

File: Detectors/MUON/MID/Raw/test/testRaw.cxx [hunk 1]
```diff
@@ -137,6 +137,7 @@ BOOST_AUTO_TEST_CASE(ColumnDataConverter)
 
   std::vector<o2::mid::ROFRecord> rofs;
   std::vector<o2::mid::LocalBoardRO> outData;
+
```
Comment (matthiasrichter) [inline Detectors/MUON/MID/Raw/test/testRaw.cxx:140]: consider removing the inserted blank line

Comment (aalkin) [reply Detectors/MUON/MID/Raw/test/testRaw.cxx:140]: Most likely due to merge conflicts, I will check.

**Comment by davidrohr** [comment]
@aalkin : There is some problem in the GPU compilation, which might now know std::size_t.

**Comment by aalkin** [comment]
> There is some problem in the GPU compilation, which might now know std::size_t.
@davidrohr Can you point me to the particular files where it will be a problem? I cannot test GPU compilation locally.

**Comment by davidrohr** [comment]
@aalkin : In principle, you can build in the GPU CI container to test GPU builds.

Files used for GPU compilation will be everything in the GPU folder and I guess all files including a GPUCommon???.h file.

An alternative might be to put just a typedef for size_t in the std namespace if needed.
What is actually the reason behind this change?

**Comment by aalkin** [comment]
> What is actually the reason behind this change?
Strictly speaking already C++11 deprecates `::size_t` in global namespace (wtih `stddef.h`), though it was not enforced by most compilers. However, in gcc10 now `<stdexcept>` is not included in some of the standard headers which cascades in some cases to `size_t` (and some other things) not being available as well. With `std::size_t` (strict C++ standard) we can avoid relying on compilers to maintain compatibility with deviations. Alternatively one can bring `size_t` into global namespace with `using std::size_t` which you still would have to put in many places.

**Pull request closed without merging**